### PR TITLE
[Feat][Spark] Add examples to show how to load/dump data from/to GraphAr for Nebula

### LINKED ISSUE #34: [Feat] Integrate NebulaGraph spark connector as input data source for GraphAr spark tool
**Is your feature request related to a problem? Please describe.**
The graph data migration between NebulaGraph and GraphAr could be an important application of GraphAr. This can be  implemented based on the NebulaGraph Spark connector and the GraphAr Spark library, including reading graph data from NebulaGraph to generate GAR files, and reading from GraphAr to create/update instances in NebulaGraph.

**Describe the solution you'd like**
Please refer to the integration with Neo4j (#107).

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by liuxiaocs7**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

close #34
example in gar-test: https://github.com/GraphScope/gar-test/pull/4

## Types of changes

What types of changes does your code introduce to GraphAr?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/alibaba/GraphAr/blob/main/CONTRIBUTING.rst) doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wey-gu** [comment]
Dear Great Creators of GraphAr Project!
This is wey from the NebulaGraph community, we are so very excited to have a chance to enable GraphAr - NebulaGraph connection in Hacktoberfest, I wonder if it's possible to request your support on:

- Adding hactoberfest-accepted label to this PR?
- Help review and hopfully collaborate to merge this change in October?

This will enable the contributor to be part of the Hactoberfest project and it means a lot!

Thanks and appreciate your help!

Best Regards, Wey

**Comment by lixueclaire** [comment]
> Dear Great Creators of GraphAr Project! This is wey from the NebulaGraph community, we are so very excited to have a chance to enable GraphAr - NebulaGraph connection in Hacktoberfest, I wonder if it's possible to request your support on:
> 
> * Adding hactoberfest-accepted label to this PR?
> * Help review and hopfully collaborate to merge this change in October?
> 
> This will enable the contributor to be part of the Hactoberfest project and it means a lot!
> 
> Thanks and appreciate your help!
> 
> Best Regards, Wey

Dear Wey,

Thank you for reaching out to us and expressing your excitement about enabling the GraphAr - NebulaGraph connection in Hacktoberfest. We appreciate your interest in our project and we are happy to support you.
We would be delighted to review your PR and collaborate with you to merge the changes in October. 
If you have any further questions or need assistance, please feel free to reach out to us.

Best regards,
Xue

**Comment by acezen** [comment]
The `hactoberfest-accepted ` label has been added to the PR.

Thanks for the EXCITING work to enabling the feature we expect. As @lixueclaire said, we are delighted to review the PR and collaborate with the Hactoberfest project.

**Comment by liuxiaocs7** [comment]
> The `hactoberfest-accepted ` label has been added to the PR.
> 
> Thanks for the EXCITING work to enabling the feature we expect. As @lixueclaire said, we are delighted to review the PR and collaborate with the Hactoberfest project.

Thanks for your help, but I confirmed that the label should be `hacktoberfest-accepted` rather than `hactoberfest-accepted`. Could you help modify it in your free time? 

CC: @wey-gu

**Comment by wey-gu** [comment]
> > Dear Great Creators of GraphAr Project! This is wey from the NebulaGraph community, we are so very excited to have a chance to enable GraphAr - NebulaGraph connection in Hacktoberfest, I wonder if it's possible to request your support on:
> > 
> > * Adding hactoberfest-accepted label to this PR?
> > * Help review and hopfully collaborate to merge this change in October?
> > 
> > This will enable the contributor to be part of the Hactoberfest project and it means a lot!
> > Thanks and appreciate your help!
> > Best Regards, Wey
> 
> Dear Wey,
> 
> Thank you for reaching out to us and expressing your excitement about enabling the GraphAr - NebulaGraph connection in Hacktoberfest. We appreciate your interest in our project and we are happy to support you. We would be delighted to review your PR and collaborate with you to merge the changes in October. If you have any further questions or need assistance, please feel free to reach out to us.
> 
> Best regards, Xue

Wow, thanks so very much for the prompt help, care and powerful support!

**Comment by wey-gu** [comment]
> The `hactoberfest-accepted ` label has been added to the PR.
> 
> Thanks for the EXCITING work to enabling the feature we expect. As @lixueclaire said, we are delighted to review the PR and collaborate with the Hactoberfest project.

Dear @acezen

Wow, many thanks for the super-fast and warm support!!!

**Comment by wey-gu** [comment]
> > The `hactoberfest-accepted ` label has been added to the PR.
> > Thanks for the EXCITING work to enabling the feature we expect. As @lixueclaire said, we are delighted to review the PR and collaborate with the Hactoberfest project.
> 
> Thanks for your help, but I confirmed that the label should be `hacktoberfest-accepted` rather than `hactoberfest-accepted`. Could you help modify it in your free time?
> 
> CC: @wey-gu

OMG, I'm so very sorry/ashamed to provide a typo on the label... let's wait @acezen for another go 🙇

File: spark/README.md [hunk 1]
```diff
@@ -135,12 +135,86 @@ echo "match (a) -[r] -> () delete a, r;match (a) delete a;" | cypher-shell -u ${
...
 
 Then run the example:
+
 ```bash
 scripts/run-graphar2neo4j.sh
 ```
 
 The example will import the movie graph from GraphAr to Neo4j and you can check the result in the Neo4j browser.
 
+## Running NebulaGraph to GraphAr example
```
Comment (acezen) [inline spark/README.md:145]: The Nebula2GraphAr example needs docker, we need to proved a hint here to let users know they should install docker first

**Review by acezen — APPROVED** [review]
LGTM, please rebase to main.
Thanks for the job.

**Comment by liuxiaocs7** [comment]
> hi, @liuxiaocs7, I not sure the PR can close the issue [vesoft-inc/nebula-spark-connector#76](https://github.com/vesoft-inc/nebula-spark-connector/issues/76) because it's just an example to exchange a specific data(basketballplayergraph graph) between nebula and GraphAr, the example can't use for other data. Do you think this can implement as a general tool that fits for every graph?
> 
> BTW, @wey-gu @lixueclaire, do you have some suggestions for that?

Sure, thank you for your suggestion, I will remove `close` it and consider how to implement a more general tool in other PRs!

**Review by lixueclaire — APPROVED** [review]
LGTM~ Thanks for your contribution. Thanks for your contribution. It would be even better if you could update the [documentation](https://alibaba.github.io/GraphAr/applications/spark.html) or create an issue about it.

**Review by lixueclaire — APPROVED** [review]
LGTM~ Thanks for your contribution.  It would be even better if you could update the [documentation](https://alibaba.github.io/GraphAr/applications/spark.html) or create an issue about it.

**Comment by lixueclaire** [comment]
> > hi, @liuxiaocs7, I not sure the PR can close the issue [vesoft-inc/nebula-spark-connector#76](https://github.com/vesoft-inc/nebula-spark-connector/issues/76) because it's just an example to exchange a specific data(basketballplayergraph graph) between nebula and GraphAr, the example can't use for other data. Do you think this can implement as a general tool that fits for every graph?
> > BTW, @wey-gu @lixueclaire, do you have some suggestions for that?
> 
> Sure, thank you for your suggestion, I will remove `close` it and consider how to implement a more general tool in other PRs!

Sounds good! Maybe you can refer to #243 for this.

**Comment by acezen** [comment]
@liuxiaocs7 the example data https://github.com/GraphScope/gar-test/pull/4 has been merge, you can update the testing submodule to  latest commit.

**Comment by liuxiaocs7** [comment]
> LGTM~ Thanks for your contribution. It would be even better if you could update the [documentation](https://alibaba.github.io/GraphAr/applications/spark.html) or create an issue about it.

Issue created here: https://github.com/alibaba/GraphAr/issues/248, i'll finish ASAP.

**Comment by liuxiaocs7** [comment]
> @liuxiaocs7 the example data [GraphScope/gar-test#4](https://github.com/GraphScope/gar-test/pull/4) has been merge, you can update the testing submodule to latest commit.

Hi, @acezen, should i follow this to update `gar-test` submodule: https://stackoverflow.com/questions/5828324/update-git-submodule-to-latest-commit-on-origin

**Comment by wey-gu** [comment]
Amazing OpenSource collaboration! I would like to express my utmost respect, appreciation, and admiration!

**Comment by wey-gu** [comment]
> > @liuxiaocs7 the example data [GraphScope/gar-test#4](https://github.com/GraphScope/gar-test/pull/4) has been merge, you can update the testing submodule to latest commit.
> 
> Hi, @acezen, should i follow this to update `gar-test` submodule: https://stackoverflow.com/questions/5828324/update-git-submodule-to-latest-commit-on-origin

@liuxiaocs7  think of submodule a way to let the parent module track submoudle as a folder only on its git hash.

With that in mind, when upating a submodle, we could just do the needed pull(to ensure its hash is the one from the remote) and do git add as a commit on the parent module.

What git submodule update was just to pull submodule based on its tracked git hash.

**Pull request merged**